### ISSUE TITLE
[Question] PPO does not learn with VecNormalize as opposed to "naive" normilization

### ISSUE THREAD

**Issue by philippwulff** [DLR-RM/stable-baselines3#416]
I am experiencing some strange behavior with the `VecNormalize` wrapper class, as it does not improve the learning of my PPO agent. I tried normalizing the observation values manually before that (I know the highs and lows of the observation space) and that worked really well. My own normalization just used 

```
(rl_observations - self.observation_space.low) / (self.observation_space.high - self.observation_space.low + 0.001) * 2 - 1
```

This produces observations between -1 and 1 and the agent learned the task quite well with them. However, when using `VecNormalize` the agent does not learn much.

I checked that the normalization statistics are the same for the training and evaluation environment using `env.obs_rms.__dict__`:

```
{'mean': array([8.01119052e+00, 9.78909299e-02, 1.09663139e+01, 1.21429955e+01,
        1.34257686e+02, 7.16278853e+01, 4.57971291e+01, 0.00000000e+00,
        5.04671978e+01, 1.49221039e+02, 1.40158012e+02]),
 'var': array([9.96172994e+00, 5.45141906e-01, 9.31504417e+00, 1.26612605e+01,
        1.27781424e+03, 1.39235962e+03, 1.38437438e+02, 8.33244454e-12,
        1.63296507e+03, 7.93307883e+01, 9.50489127e+02]),
 'count': 12001280.0001}
```

Maybe it does not work, because the way that the stable baselines 3 wrapper class normalizes is by calculating `(obs - mean) / std` which is different to what I am doing. 

This is the training loss when wrapping the env with `VecNormalize`:
![Screenshot from 2021-05-03 18-28-03](https://user-images.githubusercontent.com/45896065/116903901-57be3d80-ac3d-11eb-885e-08ae1ca1742a.png)

This is the training loss when wrapping the env with my naive normalization wrapper:
![Screenshot from 2021-05-03 20-13-50](https://user-images.githubusercontent.com/45896065/116915289-1bdea480-ac4c-11eb-9265-2d9c0cbbe900.png)

The loss is a lot better and when visually observing the environment, the agent also acts more optimal. Why does `VecNormalize` not perform well?

**Comment by Miffyli** [comment]
@araffin can you shed light on this? I assume this is one of those "you have to try it for each environment" kind of things, or is static normalization always better (if it can be done)?

**Comment by araffin** [comment]
The loss won't tell you much... You need to check mean episodic (unnormalized) reward. `VecNormalize` also normalizes the reward by default (scaling it).

Also, seems related to custom env, please fill the associated issue template, we are missing infos...

**Comment by philippwulff** [comment]
I had an issue with the reward function where it would sometimes return `inf` values. I resolved this and tested `VecNormalize` again. It is working well now! Thanks!
